### PR TITLE
Mntor 1188/breaches email count

### PR DESCRIPTION
**Pull request by toufali**
# References: 
Jira: MNTOR-1188

# Description
With this PR:
- the "monitored email" count in the header panel should only display verified email count
- the “Add email address” link in the header panel should open the “Settings” page instead of the “Add email” dialog
- the “Add email address” buttons in the Breaches table “zero-state” should continue to open the “Add email” dialog
- the “Add email address” buttons in the Breaches table “zero-state” should dynamically disappear after the user has reached their limit

TODO: Waiting for UX decision on whether the header link should read "manage emails" instead of "add email address".

# Screenshot (if applicable)

Screen Shot 2023-02-16 at 9 15 54 PM

# How to test
Visit the breaches page and test the acceptance criteria above.

# Checklist (Definition of Done)
- [ ] Localization strings (if needed) have been added.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.
- [ ] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

**Comment by toufali** [comment]
tagging @rhelmer for last person who worked on settings
tagging @mansaj for insight on error handling audit

**Review by Vinnl — APPROVED** [review]
I think related to your suggestion to use "Manage emails", but the link to settings wasn't visible at all when I had reached my limit, so I had to know where to go to remove an email or resend a verification email:

image

I was also momentarily confused by the "4 out of 5" - why weren't all five addresses monitored? Having a link to click through and then see that one was unverified will hopefully help there.

But the code does what it should do, and nothing in particular jumps out to me, so nice job :+1:

File: src/controllers/breaches.js [hunk 1]
```diff
@@ -10,14 +10,16 @@ import { generateToken } from '../utils/csrf.js'
 import { getAllEmailsAndBreaches } from '../utils/breaches.js'
 
 async function breachesPage (req, res) {
-  const emailCount = 1 + (req.user.email_addresses?.length || 0) // +1 because user.email_addresses does not include primary
   // TODO: remove: to test out getBreaches call with JSON returns
   const breachesData = await getAllEmailsAndBreaches(req.user, req.app.locals.breaches)
+  const emailVerifiedCount = breachesData.verifiedEmails?.length ?? 0
```
Comment (flozia) [inline src/controllers/breaches.js:15]: Now we decided to not include unverified emails towards the limit we’d also need to pass the `emailVerifiedCount` to the settings page layout. Currently, we use the [total count for disabling the “Add email address” button](https://github.com/mozilla/blurts-server/blob/MNTOR-1188/breaches-email-count/src/views/partials/settings.js#L130) in the list of monitored emails.

**Comment by toufali** [comment]
> ...but the link to settings wasn't visible at all when I had reached my limit

Agreed @Vinnl – I think we need more clarity from UX/PM.  I will add you to the Slack thread where I have been groaning on about this.  Specific to your point, I think we should change the copy to "Manage emails" and keep it on screen at all times.  (We currently hide "Add email" as you noted, because a user has already reached their limit.)

File: src/controllers/breaches.js (hunk 1 above)
Comment (toufali) [reply src/controllers/breaches.js:15]: I might have lost a thread (or just confused) @flozia – the CTAs for this page behave as though a user has a hard limit regardless of verified/unverified.  For example, if a user has 3 verified emails and 2 unverified emails, they would not be allowed to add another email, and the add-email CTAs are hidden.  So at least in regards to this code, unverified emails _do_ count towards the limit, and I believe this is in line with Settings page?

The difference is with the "3 of 5 emails monitored" copy – this only includes verified emails.  And the CTA below it seems wrongly worded – I'll add you to the UX/PM conversation as well.

**Review by flozia — APPROVED** [review]

**Review by flodolo — APPROVED** [review]

**Pull request merged**